keras_core/backend/__init__.py

```diff
@@ -1,4 +1,10 @@
-from keras_core.backend.config import backend
+from keras_core.backend.config import backend, _SUPPORTED_BACKENDS
+
+if backend() not in _SUPPORTED_BACKENDS:
```

Comment:
I don't think the changes related to `_SUPPORTED_BACKENDS` add value here, since you can't add a new backend just by editing the list (you need to edit this file first anyway) and there's no reduction of redundancy. Please revert.